c_src/erl_optics.c

```diff
@@ -286,10 +378,136 @@ static ERL_NIF_TERM eo_optics_free(
     return atom_ok;
 }
 
+struct map_ctx {
+  ErlNifEnv *env;
+  ERL_NIF_TERM map;
+};
+
+static void backend_eo (void *ctx, enum optics_poll_type type, const struct optics_poll *poll)
+{
+  if (type != optics_poll_metric) return;
+
+  struct map_ctx *m = (struct map_ctx *) ctx;
+  
+  ERL_NIF_TERM key = enif_make_string(m->env, poll->key, ERL_NIF_LATIN1);
+  ErlNifBinary key_bin;
+  enif_alloc_binary(strlen(poll->key), &key_bin);
+  enif_term_to_binary(m->env, key, &key_bin);
+
+  //this removes the first 4 bytes of the binary string, which are
```

Comment:
1. Why? The comment explains what the code does, rather than explaining why it's being done in the first place.
2. Would it be possible to use [this](http://erlang.org/doc/man/erl_nif.html#enif_make_new_binary) function instead?

Reply:
Good idea. Done.